modules/oe_content_persistent/tests/src/Kernel/MediaPurlMatcherTest.php

```diff
@@ -0,0 +1,125 @@
+      'id' => 'test',
+      'description' => 'Test type.',
+      'source' => 'file',
+    ]);
+    $media_type->save();
+    $source_field = $media_type->getSource()->createSourceField($media_type);
+    $source_field->getFieldStorageDefinition()->save();
+    $source_field->save();
+    $media_type->set('source_configuration', [
+      'source_field' => $source_field->getName(),
+    ])->save();
+
+    // Linkit doesn't care about the actual resource, only the entity.
+    foreach (['gif', 'jpg', 'png'] as $ext) {
+      $file = File::create([
+        'uid' => 1,
+        'filename' => 'image-test.' . $ext,
+        'uri' => 'public://image-test.' . $ext,
+        'filemime' => 'text/plain',
+        'status' => FILE_STATUS_PERMANENT,
+      ]);
+      $file->save();
+
+      $media = Media::create([
+        'bundle' => 'test',
+        $source_field->getName() => ['target_id' => $file->id()],
+      ]);
+      $media->save();
+    }
+
+    // Create user 1 who has special permissions.
+    \Drupal::currentUser()->setAccount($this->createUser(['uid' => 1]));
+  }
+
+  /**
+   * Tests media PURL matcher.
+   */
+  public function testMediaMatcherWithDefaultConfiguration() {
+    $base_url = $this->config('oe_content_persistent.settings')->get('base_url');
+    $matcher_manager = $this->container->get('plugin.manager.linkit.matcher');
+
+    /** @var \Drupal\linkit\MatcherInterface $plugin */
+    $plugin = $matcher_manager->createInstance('entity:media', []);
+    $suggestions = $plugin->execute('image-test');
+    $this->assertEquals(3, count($suggestions->getSuggestions()), 'Correct number of suggestions.');
+
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $media_storage */
+    $media_storage = $this->container->get('entity_type.manager')->getStorage('media');
+    // Verify suggestion paths.
+    foreach ($suggestions->getSuggestions() as $key => $suggestion) {
+      $media = $media_storage->load($key + 1);
+      $this->assertEquals($base_url . $media->uuid(), $suggestion->getPath());
+    }
+
+    /** @var \Drupal\linkit\MatcherInterface $plugin */
+    $plugin = $matcher_manager->createInstance('entity:media', [
+      'settings' => [
+        'thumbnail' => [
+          'show_thumbnail' => TRUE,
+          'thumbnail_image_style' => 'linkit_result_thumbnail',
+        ],
+      ],
+    ]);
+    $suggestions = $plugin->execute('image-test');
+    $this->assertEquals(3, count($suggestions->getSuggestions()), 'Correct number of suggestions.');
+
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $media_storage */
+    $media_storage = $this->container->get('entity_type.manager')->getStorage('media');
+    // Verify suggestion paths.
+    foreach ($suggestions->getSuggestions() as $key => $suggestion) {
+      $media = $media_storage->load($key + 1);
+      $this->assertEquals($base_url . $media->uuid(), $suggestion->getPath());
+      $this->assertContains('linkit_result_thumbnail', $suggestion->getDescription());
```

Comment:
Arent we rendering more that we can assert here?

Reply:
There can be more things like the metadata but I don't think its in the scope of this class. Or do you mean I should be more precise with what I should check?